components/InputBox/index.tsx

```diff
@@ -673,13 +720,147 @@ const InputBox = ({
+      if (conversations.length > 15) {
+        const newPage = page + 1;
+        setPage(newPage);
+        loadData(newPage);
+      }
+    }
+  };
+
+  //pagination loader in the footer
+  const renderFooter = () => {
+    return isLoading ? (
+      <View style={{paddingVertical: 20}}>
+        <ActivityIndicator size="large" color={STYLES.$COLORS.SECONDARY} />
+      </View>
+    ) : null;
+  };
+
+  const handleInputChange = async (e: any) => {
+    if (chatRequestState === 0 || chatRequestState === null) {
+      if (e.length >= MAX_LENGTH) {
+        dispatch({
+          type: SHOW_TOAST,
+          body: {
+            isToast: true,
+            msg: CHARACTER_LIMIT_MESSAGE,
+          },
+        });
+      } else if (e.length < MAX_LENGTH) {
+        setMessage(e);
+        setFormattedMessage(e);
+      }
+    } else {
+      let modifiedMessage = deleteRouteIfAny(e, message);
+      setMessage(modifiedMessage);
+      setFormattedMessage(modifiedMessage);
+
+      const newMentions = detectMentions(e);
+
+      if (newMentions.length > 0) {
+        const length = newMentions.length;
+        setTaggedUserName(newMentions[length - 1]);
+      }
+
+      //debouncing logic
+      clearTimeout(debounceTimeout);
+
+      let len = newMentions.length;
+      if (len > 0) {
+        const timeoutID = setTimeout(async () => {
+          setPage(1);
+          const res = await taggingAPI({
+            page: 1,
+            searchName: newMentions[len - 1],
+            chatroomId: chatroomID,
+            isSecret: false,
+          });
+          if (newMentions.length > 0) {
+            let groupTagsLength = res?.group_tags?.length;
+            let communityMembersLength = res?.community_members.length;
+            let arrLength = communityMembersLength + groupTagsLength;
+            if (arrLength > 5) {
+              setUserTaggingListHeight(5 * 58);
+            } else if (arrLength < 5) {
+              let height = communityMembersLength * 58 + groupTagsLength * 80;
+              setUserTaggingListHeight(height);
+            }
+            setUserTaggingList(res?.community_members);
+            setGroupTags(res?.group_tags);
+            setIsUserTagging(true);
+          }
+        }, 500);
+
+        setDebounceTimeout(timeoutID);
+      } else {
+        if (isUserTagging) {
+          setUserTaggingList([]);
+          setGroupTags([]);
+          setIsUserTagging(false);
+        }
+      }
+    }
+  };
   // this function is for editing a conversation
   const onEdit = async () => {
-    let selectedConversation = selectedMessages[0];
+    let selectedConversation = editMessage;
     let conversationId = selectedConversation?.id;
     let previousConversation = selectedConversation;
-    let editedConversation = message;
+
     let changedConversation;
+    let conversationText = replaceMentionValues(message, ({id, name}) => {
+      if (name === '@participants' || name === '@everyone') {
```

Comment:
why hard checks for group tags, if we create more tags in future, then we need a separate release for ReactNative as well.